eth/v1alpha1/validator.proto

```diff
@@ -15,13 +15,119 @@ syntax = "proto3";
+    rpc Block(BlockRequest) returns (Block) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/block"
+        };
+    }
+
+    // Sends the newly signed beacon block to beacon node to be broadcasted to the rest of the network.
+    //
+    // The validator sends the newly signed beacon block to the beacon node so the beacon block can
+    // be included in the beacon chain. The beacon node is expected to validate and process the
+    // beacon block into its state, however if the beacon block fails validator, it will still
+    // be broadcasted to the rest of the network but a different status code is returned.
+    rpc ProposeBlock(Block) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/eth/v1alpha1/validator/block"
+        };
+    }
+
+    // Retrieves the latest valid attestation to be attested on the beacon chain.
+    //
+    // The server returns the latest valid attestation data which represents the correct vote
+    // for the head of the beacon chain,
+    rpc Attestation(AttestationRequest) returns (AttestationData) {
+        option (google.api.http) = {
+            get: "/eth/v1alpha1/validator/attestation"
+        };
+    }
+
+    // Sends the newly signed attestation to beacon node to be broadcasted to the rest of the network.
+    //
+    // The validator sends the newly signed attestation to the beacon node for the attestation to
+    // be included in the beacon chain. The beacon node is expected to validate and aggregate the
+    // attestations into the state, if the attestation fails validator, it'd still be broadcasted but a
+    // different status code is returned.
+    rpc ProposeAttestation(Attestation) returns (google.protobuf.Empty) {
+        option (google.api.http) = {
+            post: "/eth/v1alpha1/validator/attestation"
+        };
+    }
+}
+
+message DutiesRequest {
+    // Epoch at which validators should perform their duties.
+    uint64 epoch = 1;
+    // Array of byte encoded BLS public keys.
+    repeated bytes public_keys = 2;
+}
+
+message DutiesResponse {
+    repeated Duty duties = 1;
+    message Duty {
+        bytes pub_key = 1;
+        // Slot at which a validator must attest.
+        uint64 attestation_slot = 2;
+        // Shard at which a validator must attest.
+        uint64 attestation_shard = 3;
+        // Slot at which a validator must propose on beacon chain,
+        // when returns 0, the block production is not required, meaning
+        // the validator did not get assigned to be a proposer for the
+        // input epoch.
+        uint64 block_proposal_slot = 4;
+    }
+}
+
+message BlockRequest {
+    // Slot for which the block should be proposed.
+    uint64 slot = 1;
+    // Validator's randao reveal secret of the current epoch.
+    bytes randao_reveal = 2;
+}
+
+message AttestationRequest {
+    // The proof of custody bit reported by the requesting validator. To be used in phase 1.
```

Comment:
Is this just a single bit? i.e. a boolean?

Please specify the size `1 byte`, etc.